### DISCUSSION TITLE
Using async_scope::spawn with a custom allocator

### DISCUSSION THREAD

**Discussion by haggaie** [NVIDIA/stdexec#1428] [Q&A]
Hi,

Is it possible to use `async_scope::spawn` and allocating the operation state using a custom allocator? I tried passing an environment that overrides the `get_allocator` query but it seems to be ignored.

Looking at the code, it seems that spawn always uses `new` to allocate the operation state. See [here](https://github.com/NVIDIA/stdexec/blob/35e8941066ad5c69bcb82741914585a8611917c2/include/exec/async_scope.hpp#L787).

Thanks,
Haggai

**Comment by dietmarkuehl** [comment]
This comment doesn't address the question and only outlines what the plans for a standard scope are:

The [description of `spawn()`](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3149r6.html#executionspawn) in [P3149r6](http://wg21.link/P3149R6) does state that `spawn(sender, token, env)` would use the allocators associated with `env`. It isn't particular detailed in how exactly the allocator would be used. The description is a specification, yet. The `spawn()` in [P3149r6](http://wg21.link/P3149R6) isn't a member function but an algorithm linked to an actual scope via the `token`.

**Reply by kcexn** [reply]
Has there been any progress in the last 11 months in terms of an example implementation of async_scope as described in P3149?

**Comment by haggaie** [comment]
Thanks, I noticed that spawn is implemented as a method here and not an algorithm like in P3149r6, but I didn't realize that there are still questions on how allocators would be used.
